test/test_satencoder.cpp

```diff
@@ -25,6 +25,18 @@ TEST_F(SatEncoderTest, CheckEqualWhenEqualRandomCircuits) {
   EXPECT_EQ(result, true);
 }
 
+TEST_F(SatEncoderTest, CheckErrorWhenEmpty) {
+  std::random_device        rd;
+  qc::RandomCliffordCircuit circOne(1, 0, 0);
+  circOne.erase(circOne.begin(), circOne.end());
+
+  qc::RandomCliffordCircuit circTwo(1, 0, 0);
+  circTwo.erase(circTwo.begin(), circTwo.end());
+  SatEncoder satEncoder;
+  bool       result = satEncoder.testEqual(circOne, circTwo);
```

Comment:
I am not 100% sure what this test tries to accomplish.
It creates two empty Clifford circuits (since the `depth` parameter is `0`), then it tries to delete something from the circuits (but there is nothing to delete) and then it tests the resulting (empty) circuits for equality.
If I got that scenario correctly, then this can for sure be simplified (see below) and, additionally, should return `true`, right? two empty circuits are equivalent. But maybe I got something wrong.

in any case, if I am right, then the following might be a more concise way to write this test:
```c++
const auto qc = qc::QuantumComputation(1);
SatEncoder encoder{};
const auto result = encoder.testEqual(qc, qc);
EXPECT_TRUE(result);
```